src/download-artifact.ts

```diff
@@ -74,6 +86,52 @@ export async function run(): Promise<void> {
     )
 
     artifacts = [targetArtifact]
+  } else if (isDownloadByIds) {
+    core.info(`Downloading artifacts by ID`)
+
+    const artifactIdList = inputs.artifactIds
+      .split(',')
+      .map(id => id.trim())
+      .filter(id => id !== '')
+
+    if (artifactIdList.length === 0) {
+      throw new Error(`No valid artifact IDs provided in 'artifact-ids' input`)
+    }
+
+    core.debug(`Parsed artifact IDs: ${JSON.stringify(artifactIdList)}`)
+
+    // Parse the artifact IDs
+    artifactIds = artifactIdList.map(id => {
+      const numericId = parseInt(id)
+      if (isNaN(numericId)) {
+        throw new Error(`Invalid artifact ID: '${id}'. Must be a number.`)
+      }
+      return numericId
+    })
+
+    // We need to fetch all artifacts to get metadata for the specified IDs
+    const listArtifactResponse = await artifactClient.listArtifacts({
+      latest: true,
+      ...options
+    })
```

Comment:
This should be okay to list all, since it's limited to the artifacts for this run. Within the toolkit we do have an `idFilter` option, but it's only for a singular ID:

https://github.com/actions/toolkit/blob/1b1e81526b802d1d641911393281c2fb45ed5f11/packages/artifact/src/generated/results/api/v1/artifact.ts#L176-L181

Reply:
So do you think it would be better to use what I currently have, or to write a loop that looks at each provided ID and then do an API call with `idFilter` to fetch it?

Reply:
Could we do both?

If there's a single ID - fetch it directly.
If there are multiple - list and then filter

Reply:
@joshmgross I think that would be a fine idea! I'll push some commits shortly.

Reply:
We chatted in Slack a bit and the result was that this current implementation (see commits below) is fine for now.

Perhaps one day a `getArtifactByID` (public method) will exist and we can just use that rather than making a `getArtifact` call and then checking the resulting ID.

For now this should do.